projects/fsastorefrontlib/src/cms-components/checkout/components/personal-details-navigation/personal-details-navigation.component.ts

```diff
@@ -68,7 +69,10 @@ export class PersonalDetailsNavigationComponent implements OnInit, OnDestroy {
             return this.formService.getSubmittedForm().pipe(
               map(formData => {
                 if (formData && formData.content) {
-                  if (!isProductConfigurable) {
+                  if (
+                    !isProductConfigurable &&
+                    !formGroup.get('personalDetails.city').disabled
```

Comment:
Can we execute UserAddressService.getAddresses and than check if saved address contains city/town?